### PR TITLE
Homework 3  for Lesson 3: Functions

### PR DESCRIPTION
**Pull request by chatieb**
1) Fork https://github.com/csinn/CSharp-From-Zero-To-Hero/tree/Chapter1/Homework/3
2) Take homework 2 code, place it in the forked branch and refactor it using functions. There should be as little duplicate code as possible (there should be functions for:
- Calculating BMI (weight comes in kg, height comes in meters),
- Prompt for input and converting it to int (print message for request, read console input and return converted input to int),
- Prompt for input and converting it to string (print message for request, read console input and return input),
- Prompt for input and converting it to float (print message for request, read console input and return converted input to float).
3) Put all the function you made into the right places in Checks.cs class. Run tests, make sure you pass all the tests
4) Put all program.cs logic to Lesson3.cs. Call it from main function. Program class should look like this:

public class Program
{
  static void Main(string[] args)
  {
    Lesson3.Demo();
  }
}

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4855470)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4855470&bid=16106772)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4855470)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4855470&bid=16106772)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            const int NumOfPerson = 2;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Local variables should be camel case. Regardless of being a const or not.

File: Src/BootCamp.Chapter/Checks.cs [hunk 4]
```diff
@@ -17,25 +17,24 @@ public static class Checks
         public static int PromptInt(string message)
         {
             // To do: call your implementation.
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: Remove such comments if you've done the TODO. :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+        public static float PromptFloat(string message)
+        {
+            return (Convert.ToSingle(PromptString(message)));
+        }
+
+        public static float CalculateBmi(float weightKg, float heightM)
+        {
+            //BMI is weight (kg) / [height (m)] ^ 2
+            return ((float)(weightKg / Math.Pow(heightM, 2)));
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: You can spare the outer ( ) of those return statements. They do not affect anything.

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 3 above)
Comment (chatieb) [reply Src/BootCamp.Chapter/Lesson3.cs]: It seems like people do things differently, and I must admit, it's a blur to me.  I will keep in mind to do numOfPerson for constant for this class.  I was interested in seeing if Codacy (which I never heard of until this Bootcamp) would suggest any better way of doing it.  I recently saw somebody else wanting NUM_OF_PERSON for constant.  I went to Microsoft site and they did had a "const double GrativationalConstant" at https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/const, and so, I ended up going with NumOfPerson.  I will keep in mind to do camel case for constants.  Thank you.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4855470&bid=16106772)

File: Src/BootCamp.Chapter/Lesson3.cs (hunk 3 above)
Comment (chatieb) [reply Src/BootCamp.Chapter/Lesson3.cs]: While making suggested fixes, I happen to notice that in Lesson3Tests.cs that came with the original code, there is a a line that has "const string PromptMessage".  However, I went ahead a made change to the constant that I had coded to be camel case. instead.

**Pull request closed without merging**